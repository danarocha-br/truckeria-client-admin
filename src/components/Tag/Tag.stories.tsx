import { Story, Meta } from '@storybook/react/types-6-0';

import Tag from '.';

export default {
  title: 'Components/Tag',
  component: Tag,
  parameters: {
    layout: 'centered',
  },
  args: {
    label: 'I am a tag',
    loading: false,
    isEmpty: false,
  },
  argTypes: {
    label: {
      table: {
        category: 'Modifiers',
      },
    },
    isEmpty: {
      table: {
        category: 'Modifiers',
      },
    },
    loading: {
      table: {
        category: 'Modifiers',
      },
    },
  },
} as Meta;

const Template: Story = (args) => <Tag label="I am a tag" {...args} />;

export const Primary = Template.bind({});
