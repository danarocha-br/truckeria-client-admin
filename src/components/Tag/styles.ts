import styled, { css } from 'styled-components';
import tw from 'twin.macro';

export const Container = styled.span`
  ${({ theme }) => css`
    ${tw`
      text-sm
      font-semibold
      capitalize
      flex
      justify-center
      border
      border-solid
      rounded-xs
      px-5
      py-1
    `};
    border-color: ${theme.colors.brand};
  `}
`;
