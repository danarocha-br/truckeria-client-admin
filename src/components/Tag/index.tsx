import Skeleton from 'components/Skeleton';

import * as S from './styles';

type TagProps = {
  label: string;
  loading?: boolean;
  isEmpty?: boolean;
};

const Tag = ({ label, loading, isEmpty }: TagProps) => (
  <>
    {loading ? (
      <S.Container>
        <Skeleton width={50} />
      </S.Container>
    ) : (
      <S.Container aria-label={label}>
        {isEmpty ? <Skeleton width={50} animationOff /> : label}
      </S.Container>
    )}
  </>
);

export default Tag;
