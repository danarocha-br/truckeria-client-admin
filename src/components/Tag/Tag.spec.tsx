import { screen } from '@testing-library/react';

import { renderWithTheme } from 'utils/tests/helpers';

import Tag from '.';

describe('<Tag />', () => {
  it('should render the Tag', () => {
    const { container } = renderWithTheme(<Tag label="I am a label" />);

    expect(screen.getByLabelText(/I am a label/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
