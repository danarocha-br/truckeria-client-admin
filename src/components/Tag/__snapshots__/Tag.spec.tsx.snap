// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should render the Tag 1`] = `
.c0 {
  font-size: 0.8rem;
  font-weight: 600;
  text-transform: capitalize;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-width: 1px;
  border-style: solid;
  border-radius: 0.4rem;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
  border-color: #FF8D4D;
}

<span
  aria-label="I am a label"
  class="c0"
>
  I am a label
</span>
`;
